test/browser/test_anisotropic.c

```diff
@@ -115,7 +115,12 @@ int main(int argc, char *argv[])
       while (level < 5) {
         printf("uploading level %d: %d, %d\n", level, w, h);
         assert(!glGetError());
+#if TEST_TEXSUBIMAGE
+        glCompressedTexImage2D(GL_TEXTURE_2D, level, GL_COMPRESSED_RGBA_S3TC_DXT5_EXT, w, h, 0, w*h, NULL);
```

Comment:
> https://community.khronos.org/t/glcompressedteximage2d-and-null-data/41505

IIUC this link is saying you can call `glTexImage2D` with a null pointer to allocate the texture, then <code>gl*Compressed*TexSubImage2D</code> to initialize it.

Reply:
According https://github.com/emscripten-core/emscripten/issues/19300, GLES users expect to be able to call `glCompressedTexImage2D` with the null final argument.

So I think the test code here should use what the existing GLES code expects to be able to do.

Reply:
> IIUC this link is saying you can call `glTexImage2D` with a null pointer to allocate the texture, then `gl_Compressed_TexSubImage2D` to initialize it.

I am not sure if that is how the spec is supposed to work. Calling `TexSubImage` will not change a type of the texture that was previously created with a call to `TexImage`?

I tried with the following test code:

```html
<!DOCTYPE html><html><body><script>
var gl = document.createElement("canvas").getContext('webgl');
gl.bindTexture(gl.TEXTURE_2D, gl.createTexture());
var dxt1 = gl.getExtension('WEBGL_compressed_texture_s3tc').COMPRESSED_RGB_S3TC_DXT1_EXT;
var w = 1024;
gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA, w, w, 0, gl.RGBA, gl.UNSIGNED_BYTE, null);
var foo = new Uint8Array(w*w/2);
gl.compressedTexSubImage2D(gl.TEXTURE_2D, 0, 0, 0, w, w, dxt1, foo);
</script></body></html>
```
which gives a warning in Firefox:

![image](https://github.com/user-attachments/assets/cbf8213c-8aa6-40a3-ae46-a743053200de)

that suggests the same.